bip-hashrate-escrows.mediawiki

```diff
@@ -0,0 +1,469 @@
+|-
+| 3
+| Escrow Name/Description
+| 120
+| string
+| A human-readable name and description of the sidechain. More than enough space to hold a 32 byte hash. Helps prevent destructive interference among sidechains (see below).
+|-
+| 4
+| Critical Private Key
+| 32
+| hex
+| This is the Double-SHA256 of the binary release of the reference software. It ties the sidechain to a specific software release, and doubles as a way of signing withdrawals (see below).
+|-
+| 5
+| Critical Address*
+| 32
+| string
+| This is derived by taking f(x) = ImportPrivateKey(HexToBase58Check(x)) of field #3. It is intentionally in the human-readable format {{although it could be 25 bytes of binary}}.
+|-
+| 10
+| "CTIP" -- Part 1 "TxID"*
+| 32
+| hex
+| The CTIP, or "Critical (TxID, Index) Pair" is a variable for keeping track of where the escrow's money is (ie, which member of the UTXO set).
+|-
+| 11
+| "CTIP" -- Part 2 "Index"*
+| 4
+| hex
+| Of the CTIP, this is second element of the pair: the Index. See #10 above.
+|}
+
+\* Denotes a "convenience field", the entry for this field is derived from other fields, or from the blockchain-state itself. The escrow-creator does not need to provide these values in M1 (or anywhere).
+
+Escrow_DB requires 223 bytes [1+120+32+32+2+32+4] for each escrow in the main blockchain. Of these, 70 bytes [2+32+32+4] are merely for convenience. Therefore, a sidechain is defined (see "M1") by 153 bytes of information.
+
+====Notes on D1====
+
+# D1 will always exist.
+# D1 will always have a unique sort (first by "Escrow Number" (field #1), and second by "Active" (field #2)). There is only ever one (escrow number, Active) pair. 
+# D1 is updated according to M1 and M2 (below).
+# If a new entry is added to D1 with an "Escrow Number" that is already in use, then this entry will either eventually be removed (because it was not supported with an M2), or it will eventually overwrite the old entry (if it *was* supported via M2).
+
+
+====Notes on D1====
+
+=====Obligations Placed on Miners=====
+
+Miners have always upgraded their software according to criteria that are known only to them (in other words, "whenever they want").
+
+However, this soft fork imposes two new criteria upon them. First: miners should only upgrade their software, if any modification to the portfolio of sidechains [that are added/removed in the upgrade] can be expected to increase miner wealth. Trivially, this implies that miners should make sure that the upgrade doesn't overwrite (and destroy) an existing sidechain that they like! But, more seriously, it implies that miners should take an interest in what the sidechain is doing to the mainchain and other sidechains (see below).
```

Comment:
These requirements are subjective and cannot be imposed.

As it stands, I think this may be a technical unsoundness. (impediment to a BIP number assignment) :(

Reply:
The first requirement was an implied part of the SegWit UASF BIP.

And both are really not that subjective -- knowledge about "what will maximize profits" does exist. And even if it exists imperfectly, then that is no different from any other kind of knowledge.

And in the UASF case these "requirements" ended up being imposed, in practice.

But if you want to reject the BIP, arguing that it is technically unsound to require miners to maximize their profitability, I am fine with that.

Reply:
Profitability is subjective. For example, Evilsidechain could be offering a large bribe.

I don't understand how you're trying to tie this to UASFs... maybe I'm missing something?

Reply:
> Profitability is subjective. For example, Evilsidechain could be offering a large bribe.

Either the bribe is objectively larger than some value, or it objectively is not. So I don't see how this is an example of subjectivity?

> I don't understand how you're trying to tie this to UASFs

In the UASF, miners must decide whether they want to upgrade their software (surrender to the UASF) or not (in which case they fight the UASF). And they will make this decision based on their (imperfect) knowledge of the profitability of each version of the protocol.

In fact , this is true of all soft forks, not just UASFs. As someone *could* propose a softfork that made coinbase-txn maturity 300,000 blocks, instead of 100; and which reduced the blocksize/weight to 500 bytes. And even in the pre-UASF era, miners would have to decide whether or not to install that software.

Reply:
The bribe might not be offered to all miners equally.

It doesn't seem like you understand UASFs... miners must choose to upgrade to remain part of the network, or failure to do so would be them launching an altcoin (whether they run the old software or make other changes). There is no pre-UASF era... the earliest softforks were all UASFs.

Reply:
> It doesn't seem like you understand UASFs... miners must choose to upgrade to remain part of the network

It seems I do not. For instance, I don't understand what happens if two mutually-exclusive UASFs are released at once, and each of these are installed by exactly half of the userbase? Then miners cannot "choose to upgrade to remain part of the network", as you say, because there are now (or will be soon) two networks.

And my example scenario is not really that hard to imagine. For example, suppose that after the SegWit UASF was proposed, someone else proposed a soft fork banning OP returns from coinbase txns (for whatever reason). Or if two people tried to reassign the same unused opcode at the same time, via a UASF alone.

> There is no pre-UASF era... the earliest softforks were all UASFs.

I refer to the modern prominence of the UASF, in *contrast* to a MASF. Specifically, I am using the terms "pre-UASF era" and "before Feb 25th, 2017" interchangeably.